.github/workflows/release.yml

```diff
@@ -44,28 +44,23 @@ jobs:
         with:
           subject-path: "${{ github.workspace }}/build/aws/elastic-apm-node-lambda-layer-*.zip"
 
-      - id: docker-vars
-        name: Set up docker variables
-        run: |-
-          if [ "${{ startsWith(github.ref, 'refs/tags') }}" == "false" ] ; then
-            # for testing purposes
-            echo "tag=test" >> "${GITHUB_OUTPUT}"
-            echo "latest=test-latest" >> "${GITHUB_OUTPUT}"
-          else
-            # version without v prefix (e.g. 1.2.3)
-            echo "tag=${GITHUB_REF_NAME/v/}" >> "${GITHUB_OUTPUT}"
-            echo "latest=latest" >> "${GITHUB_OUTPUT}"
-          fi
+      - name: Extract metadata (tags, labels)
+        id: docker-meta
+        uses: docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81  # v5.5.1
+        with:
+          images: ${{ env.DOCKER_IMAGE_NAME }}
+          tags: |
+            type=raw,value=latest,prefix=test-,enable={{is_default_branch}}
```

Comment:
Doesn't this one *always* result in having a "test-" prefix, even for a real release?

Why does `enable={{is_default_branch}}` resolve to `enabled=true` in this example run: https://github.com/elastic/apm-agent-nodejs/actions/runs/8878357355/job/24373834641

```
Run docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81
  with:
    images: docker.elastic.co/observability/apm-agent-nodejs
    tags: type=raw,value=latest,prefix=test-,enable={{is_default_branch}}
  type=raw,value=latest,prefix=test-,enable=true
  type=ref,event=tag,prefix=,suffix=
  
    context: workflow
    github-token: ***
  env:
    DOCKER_IMAGE_NAME: docker.elastic.co/observability/apm-agent-nodejs
    DOCKER_USERNAME: 
    DOCKER_PASSWORD: 
```

Reply:
That's the expected behaviour for `main` but no for git tags.

I'm trying to do the same as done in

```
          if [ "${{ startsWith(github.ref, 'refs/tags') }}" == "false" ] ; then
            # for testing purposes
            echo "tag=test" >> "${GITHUB_OUTPUT}"
            echo "latest=test-latest" >> "${GITHUB_OUTPUT}"
          else
            # version without v prefix (e.g. 1.2.3)
            echo "tag=${GITHUB_REF_NAME/v/}" >> "${GITHUB_OUTPUT}"
            echo "latest=latest" >> "${GITHUB_OUTPUT}"
          fi
```

Reply:
> I'm trying to do the same as done in

1. Okay, so first we want a `{{version}}` tag on "push tag":

    ```
    type=semver,pattern={{version}}
    ```

2. Then we want a `latest` tag on "push tag". Per https://github.com/docker/metadata-action/#latest-tag this is already handled by default. 

    Ideally we only want this "latest" if this is the latest major version, like we'd had with "push-docker.sh", but we are skipping that requirement for now. I *think* we could make this explicit by using the "flavor" input. On the current latest major branch (i.e. on "main") we want the default:

    ```
    flavor: |
      latest=auto
    ```

    and on other branches (like a future "4.x" branch if/when we have a "5.x" version) we'd want:

    ```
    flavour: |
      latest=false
    ```

3. Then we want a `test` (or perhaps `test-latest` or some other name) Docker tag for pushes to a branch. (We could limit this to branch "main" or to any branch possibly used for testing when the "on:" section is edited temporarily)

    ```
    type=ref,event=branch,value=test
    ```

    However, I'm not sure that works. The `value=` argument is not documented for https://github.com/docker/metadata-action/#typeref

    An alternative might be to limit to only "wanting" to have an `edge` tag for pushes to the default branch (i.e. to "main"), then we could use https://github.com/docker/metadata-action/?tab=readme-ov-file#typeedge

    ```
    type=edge
    ```


Assuming we'd be okay with the "edge" thing, then we could have this:


```yaml
      - name: Determine Docker tags and labels
        id: docker-meta
        uses: docker/metadata-action@8e5442c4ef9f78752691e2d8f8d19755c6f78e81  # v5.5.1
        with:
          images: ${{ env.DOCKER_IMAGE_NAME }}
          flavor: |
            latest=auto
          tags: |
            # "1.2.3" and "latest" on push of tag "v1.2.3"
            type=semver,pattern={{version}}
            # "edge" tag on push to default branch
            type=edge
```

Reply:
Let me add your suggestion then

Reply:
See 479d97b8